patches/tModLoader/Terraria/ModLoader/IModType.cs

```diff
@@ -5,11 +5,16 @@ public interface IModType : ILoadable
 		///<summary>
 		/// The mod this belongs to.
 		/// </summary>
-		public Mod Mod{get;}
+		public Mod Mod { get; }
 
 		/// <summary>
 		/// The internal name of this instance.
 		/// </summary>
-		public string Name{get;}
+		public string Name { get; }
+
+		/// <summary>
+		/// => $"{Mod.Name}/{Name}"
+		/// </summary>
+		public string FullName { get; } // default Properties when??
```

Comment:
Having this as part of the interface give the possibility for wrong implementations. If we need to use this a bunch, just make an extension methods for it, that way it is effectively "sealed" (just another case where extension properties would be nice).

Reply:
If we had either extension properties or default interface implementations (which we might get before tML 1.4 if we get mods building with C#8) then we could solve this nicely. It's unlikely any modders will implement the interface and not extend the type.